community/bolt/src/test/java/org/neo4j/bolt/security/auth/BasicAuthenticationTest.java

```diff
@@ -21,88 +21,67 @@
 
 import org.hamcrest.Description;
 import org.hamcrest.TypeSafeMatcher;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import java.io.IOException;
+
 import org.neo4j.kernel.api.exceptions.Status;
-import org.neo4j.kernel.api.security.AuthSubject;
-import org.neo4j.kernel.api.security.AuthenticationResult;
-import org.neo4j.kernel.api.security.exception.InvalidAuthTokenException;
+import org.neo4j.kernel.api.security.PasswordPolicy;
+import org.neo4j.kernel.impl.security.User;
+import org.neo4j.server.security.auth.AbstractUserRepository;
 import org.neo4j.server.security.auth.BasicAuthManager;
-import org.neo4j.server.security.auth.BasicSecurityContext;
-import org.neo4j.server.security.auth.PasswordPolicy;
+import org.neo4j.server.security.auth.ListSnapshot;
 import org.neo4j.server.security.auth.UserRepository;
-import org.neo4j.time.FakeClock;
+import org.neo4j.test.rule.fs.EphemeralFileSystemRule;
+import org.neo4j.time.Clocks;
 
 import static java.util.Collections.singletonList;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertTrue;
-import static org.mockito.Matchers.anyMap;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 import static org.neo4j.helpers.collection.MapUtil.map;
 
 public class BasicAuthenticationTest
 {
     @Rule
     public ExpectedException exception = ExpectedException.none();
 
+    @Rule
+    public final EphemeralFileSystemRule fsRule = new EphemeralFileSystemRule();
+    private Authentication authentication;
+
     @Test
     public void shouldNotDoAnythingOnSuccess() throws Exception
     {
-        // Given
-        BasicAuthManager manager = mock( BasicAuthManager.class );
-        BasicSecurityContext securityContext = mock( BasicSecurityContext.class );
-        BasicAuthentication authentication = new BasicAuthentication( manager );
-        AuthSubject authSubject = mock( AuthSubject.class );
-
-        when( manager.login( anyMap() ) ).thenReturn( securityContext );
-        when( securityContext.subject() ).thenReturn( authSubject );
-        when( authSubject.getAuthenticationResult() ).thenReturn( AuthenticationResult.SUCCESS );
-
-        //Expect nothing
-
         // When
-        authentication.authenticate( map( "scheme", "basic", "principal", "bob", "credentials", "secret" ) );
+        AuthenticationResult result =
+                authentication.authenticate( map( "scheme", "basic", "principal", "mike", "credentials", "secret2" ) );
+
+        // Then
+        assertThat( result.getSecurityContext().description(), equalTo( "user 'mike' with FULL" ) );
```

Comment:
Hmm, I would prefer to assert on the subject and access-mode directly. Now the test is coupled to the user description of the security context, even though it is a simple unit test that shouldn't care about user representation.